var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -606,6 +203,14 @@ def setup_environment(self, spack_env, run_env):
             run_env.set('CC', join_path(self.spec.prefix.bin, 'clang'))
             run_env.set('CXX', join_path(self.spec.prefix.bin, 'clang++'))
 
+    # When building flang we do not use the mono repo
```

Comment:
Does this mean that `llvm+flang+anything_else` won't build properly?

Reply:
Saw the comment below. In that case, we may want to add a `conflicts` for `+flang` and `+anything_else`.

Reply:
Yeah, really not fond of having it like that, also not sure why that would be the case. Is there any reason you can’t build llvm with clang and Flang simultaneously? If so, I almost want Flang to be a separate package entirely, but at a minimum have a check for nonsensical requirements.

Reply:
Clang and Flang do use different versions of LLVM. Flang is a separate package, but the LLVM component of it is part of this package.

@trws I know what you are talking about. I raised this question a while back and I didn't get a whole lot of feedback.
https://github.com/spack/spack/issues/13117

`flang` will be superceded by `f18` in about 6-months to a year, but the time scales for upstreaming the flang-version of LLVM into the upstream clang-version of LLVM may be longer.

@homerdin may know more.

Reply:
Fair enough, we may be stuck with that then.  I’d personally prefer that move into the Flang package honestly, since the resulting LLVM isn’t usable as an LLVM, but that issue predates this PR so I’m not going to hold @homerdin to implementing it.

Reply:
Right, the Flang parts can be reworked or removed.  For this PR I'm mostly trying to move the build to use the mono-repo which is the official way to build now and greatly simplifies the package.  The functionality is largely the same, currently if you build `llvm@flang20180921+polly` you do not build polly.

While having a version that is only meant to be used by another package might be better than a variant that is, I don't think either is what we want.